server/api/Routers/OrderRouter.js

```diff
@@ -0,0 +1,34 @@
+const router = require('express').Router();
+const { Order, User } = require('../../db/models/');
+
+// Routes:
+// /api/orders (all orders)
+router.get('/orders', async (req, res) => {
+  const orders = await Order.findAll();
+  res.send(orders);
+});
+
+// /api/orders/:id (specific order, which includes products)
+router.get('/orders/:id', async (req, res) => {
+  const order = await Order.findOne({ where: { id: req.params.id } });
+  res.send(order);
+});
+
+// /api/users/:id/orders (all orders of a user)
+router.get('/users/:id/orders', async (req, res) => {
+  const user = await User.findByPk(req.params.id);
+  const orders = await user.getOrders();
+  res.send(orders);
+});
+
+// /api/users/:id/orders/:id (specific order of a user, which includes products)
+router.get('/users/:id/orders/:orderId', async (req, res) => {
+  const user = await User.findByPk(req.params.id);
+  const orders = await user.getOrders();
+  const order = orders.filter(order =>
+    order.id === req.params.orderId ? true : false
```

Comment:
do you need to add a return here? can't recall with Filter

Reply:
- try/catch with async await
- may need to return the order vs a boolean
- also if you filter there should be one match that it returns so is doing [0] redundant? I also could be reading that wrong

Reply:
You do need to return with filter. It's a one line without brackets so it returns implicitly, but I can change it to be explicit. 
```const order = orders.filter(order => { 
return order.id === req.params.orderId ? true : false;
 }``` 
You need to return a boolean with the filter cb, then it returns an array with all the values which returned true from the call back. 
There should be only one match but filter returns an array so the [0] at the end returns that one match out of the array. So that order is equal to the match instead of [the match].

Reply:
Gotcha that's cool, I think you could go further (based on some stackoverflow search) and remove the booleans completely:
```
const order = orders.filter(order => { 
return order.id === req.params.orderId
 })
```